### PR TITLE
Add configurable option for FileUpload

### PR DESCRIPTION
**Pull request by pmd1nh**
provide a configurable property for fileupload

https://github.com/OpenLiberty/open-liberty/issues/24767

### CODE REVIEW

**Comment by posmith** [comment]
L2 message review completed

**Comment by pmd1nh** [comment]
@dmuelle can you please review the messages in this PR?  Im trying to get them in for translation so they can be ready for 23.0.0.4 (translation Dcut is 3/31).  Thanks!

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -188,3 +188,6 @@ excludeAllHandledTypesClasses.desc=When this property is set to true, during sta
 
 skipEncodedCharVerification.name=Skip verification of encoded characters in URI
 skipEncodedCharVerification.desc=A request is rejected if the request URI contains the %23, %2E, %2F, or %5C encoded characters. When this property is set to true, the request URI is not checked for these encoded characters. This property is available starting in servlet 6.0, and default value is false.
+
+maxFileCount.name=The maximum number of uploaded files in a single request
+maxFileCount.desc=The maximum number of files being uploaded via the multipart/form-data request. The default is 5000.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]:
```suggestion
maxFileCount.desc=The maximum number of files that can be uploaded by a multipart/form-data request. The default is 5000.
```

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 2]
```diff
@@ -1622,3 +1622,8 @@ uri.has.fragment.character=URI has a fragment [#] character; encoded [%23] or no
 uri.has.dot.character=URI has encoded dot [%2E] character.
 uri.has.forwarslash.character=URI has encoded forward slash [%2F] character.
 uri.has.backslash.character=URI has backslash character; encoded [%5C] or not.
+
+multipart.file.count.max.exceeded=CWWWC0006E: The maximum permitted number of uploaded files, in a single request, has exceeded [{0}].
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]:
```suggestion
multipart.file.count.max.exceeded=CWWWC0006E: The number of uploaded files in the request exceeds maximum permitted number, which is [{0}].
```

Assuming [{0}] is the maximum

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 3]
```diff
@@ -1622,3 +1622,8 @@ uri.has.fragment.character=URI has a fragment [#] character; encoded [%23] or no
...
+multipart.file.count.max.exceeded.explanation=The files being uploaded via the multipart/form-data request exceeded the maximum.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]:
```suggestion
multipart.file.count.max.exceeded.explanation=The number of files that are uploaded by the multipart/form-data request exceeded the maximum limit.
```

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 4]
```diff
@@ -1622,3 +1622,8 @@ uri.has.fragment.character=URI has a fragment [#] character; encoded [%23] or no
...
+multipart.file.count.max.exceeded.useraction=Increase the max file count by setting the webContainer custom property maxFileCount in the server.xml . The default is 5000.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]:
```suggestion
multipart.file.count.max.exceeded.useraction=Increase the maximum file count by setting the maxFileCount webContainer custom property in the server.xml file. The default is maximum is 5000.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review, let me know if you have any questions, thanks

**Comment by pmd1nh** [comment]
@dmuelle I updated as per your suggestion.  Please re-review.  Thanks!

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 5]
```diff
@@ -1622,3 +1622,8 @@ uri.has.fragment.character=URI has a fragment [#] character; encoded [%23] or no
 uri.has.dot.character=URI has encoded dot [%2E] character.
 uri.has.forwarslash.character=URI has encoded forward slash [%2F] character.
 uri.has.backslash.character=URI has backslash character; encoded [%5C] or not.
+
+multipart.file.count.max.exceeded=CWWWC0006E: The number of uploaded files in the request exceeds maximum permitted number, which is [{0}].
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]:
```suggestion
multipart.file.count.max.exceeded=CWWWC0006E: The number of uploaded files in the request exceeds the maximum permitted number, which is [{0}]. 
```

Comment (dmuelle) [reply dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: @pmd1nh sorry, my suggestion introduced a typo, fixed here

**Comment by pmd1nh** [comment]
#build
        #spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
metatype name is longer than 50.  Adjusted.

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
@dmuelle I added another text about the setting of -1.  Thanks!

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 6]
```diff
@@ -188,3 +188,6 @@ excludeAllHandledTypesClasses.desc=When this property is set to true, during sta
 
 skipEncodedCharVerification.name=Skip verification of encoded characters in URI
 skipEncodedCharVerification.desc=A request is rejected if the request URI contains the %23, %2E, %2F, or %5C encoded characters. When this property is set to true, the request URI is not checked for these encoded characters. This property is available starting in servlet 6.0, and default value is false.
+
+maxFileCount.name=The maximum uploaded files in a single request
+maxFileCount.desc=The maximum number of files that can be uploaded by a multipart/form-data request. The default is 5000; setting to -1 for unlimited files upload.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]:
```suggestion
maxFileCount.desc=The maximum number of files that can be uploaded by a multipart/form-data request. The default is 5000. To enable unlimited file uploads, set this property to -1.
```

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 7]
```diff
@@ -1622,3 +1622,7 @@ uri.has.fragment.character=URI has a fragment [#] character; encoded [%23] or no
 uri.has.dot.character=URI has encoded dot [%2E] character.
 uri.has.forwarslash.character=URI has encoded forward slash [%2F] character.
 uri.has.backslash.character=URI has backslash character; encoded [%5C] or not.
+
+multipart.file.count.max.exceeded=CWWWC0006E: The number of uploaded files in the request exceeds the maximum permitted number, which is [{0}]. 
+multipart.file.count.max.exceeded.explanation=The number of files that are uploaded by the multipart/form-data request exceeded the maximum limit.
+multipart.file.count.max.exceeded.useraction=Increase the maximum file count by setting the maxFileCount webContainer custom property in the server.xml file. The default maximum is 5000; setting to -1 for unlimited files upload.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]:
```suggestion
multipart.file.count.max.exceeded.useraction=Increase the maximum file count by setting the maxFileCount webContainer custom property in the server.xml file. The default maximum is 5000.  To enable unlimited file uploads, set this property to -1. 
```

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 8]
```diff
@@ -1622,3 +1622,7 @@ uri.has.fragment.character=URI has a fragment [#] character; encoded [%23] or no
 uri.has.dot.character=URI has encoded dot [%2E] character.
 uri.has.forwarslash.character=URI has encoded forward slash [%2F] character.
 uri.has.backslash.character=URI has backslash character; encoded [%5C] or not.
+
+multipart.file.count.max.exceeded=CWWWC0006E: The number of uploaded files in the request exceeds the maximum permitted number, which is [{0}]. 
+multipart.file.count.max.exceeded.explanation=The number of files that are uploaded by the multipart/form-data request exceeded the maximum limit.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]:
```suggestion
multipart.file.count.max.exceeded.explanation=The number of files that are uploaded by the multipart/form-data request exceeded the maximum limit, which is specified by the maxFileCount webContainer custom property.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
@pmd1nh sorry for the delay, I missed the notification for your updates. Made a few suggestions, let me know if you have any questions. Thanks

**Comment by pmd1nh** [comment]
messages https://github.com/OpenLiberty/open-liberty/pull/24820

**Comment by pmd1nh** [comment]
clean up the FAT

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
All WC passed  including new tests

com.ibm.ws.fat.wc.tests.WCFileUpLoadFileCountMaxPropertyTest | 6 | 0 | 0 | 19 s | 9
-- | -- | -- | -- | -- | --
testFileUpload_overMaxFileCountWithProperty | 1 | 0 | 0 | 4 s | 1
testFileUpload_over10KWithProperty | 1 | 0 | 0 | 2 s | 2
testFileUpload_overMaxFileCountWithProperty_EE9_FEATURES | 1 | 0 | 0 | 4 s | 3
testFileUpload_over10KWithProperty_EE9_FEATURES | 1 | 0 | 0 | 2 s | 4
testFileUpload_overMaxFileCountWithProperty_EE10_FEATURES | 1 | 0 | 0 | 3 s | 5
testFileUpload_over10KWithProperty_EE10_FEATURES | 1 | 0 | 0 | 1 s | 6
 

com.ibm.ws.fat.wc.tests.WCFileUpLoadFileCountMaxTest | 6 | 0 | 0 | 15 s | 10
-- | -- | -- | -- | -- | --
testFileUpload_default5000Files | 1 | 0 | 0 | 3 s | 1
testFileUpload_overMaxFileCount | 1 | 0 | 0 | 1 s | 2
testFileUpload_default5000Files_EE9_FEATURES | 1 | 0 | 0 | 4 s | 3
testFileUpload_overMaxFileCount_EE9_FEATURES | 1 | 0 | 0 | 1 s | 4
testFileUpload_default5000Files_EE10_FEATURES | 1 | 0 | 0 | 3 s | 5
testFileUpload_overMaxFileCount_EE10_FEATURES | 1 | 0 | 0 | 1 s | 6
 

2 unrelated failed tests


no component | 92231 | 1 | 1 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.jca_fat_regr.rar.beanVal FAT tests | 41 | 0 | 1 |   |  
com.ibm.ws.request.timing_fat FAT tests | 10 | 1 | 0 |   |

**Comment by pmd1nh** [comment]
#libby

**Review by epj — APPROVED** [review]
okay

**Review by dmuelle — APPROVED** [review]

**Comment by pmd1nh** [comment]
edit: eric removed APAR ref

**Pull request merged**